lightning/backend/local.go

```diff
@@ -1613,13 +1680,20 @@ func (w *LocalWriter) writeRowsLoop() {
 			w.writeErr.Set(err)
 			return
 		}
-		if err := w.local.db.Ingest([]string{filePath}); err != nil {
-			w.writeErr.Set(err)
-			return
-		}
+		w.local.lockForImport(importMutexStateFlush)
+		err := w.local.db.Ingest([]string{filePath})
+		// The following two variable should be changed with holding mutex
```

Comment:
Seems after we `ResetEngine`, we should also reopen the `LocalWriter` 😂 need to change a lot

Reply:
No. We do not need to reopen `LocalWriter`.  Because LocalWriter holds `LocalFile` rather than `pebble.DB`. So we only need to update the field `LocalFile.db` with holding mutex.